src/lib/components/Tabs/index.js

```diff
@@ -0,0 +1,183 @@
+import {html} from "lit-element";
+import {BaseElement} from "../BaseElement";
+
+class Tabs extends BaseElement {
+  static get properties() {
+    return {
+      label: {type: String},
+    };
+  }
+
+  constructor() {
+    super();
+
+    this.prerenderedChildren = null;
+    this.tabs = null;
+
+    this.changeTab = this.changeTab.bind(this);
+    this.focusNextItem = this.focusNextItem.bind(this);
+    this.focusPreviousItem = this.focusPreviousItem.bind(this);
+    this.focusFirstItem = this.focusFirstItem.bind(this);
+    this.focusLastItem = this.focusLastItem.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren) {
+      this.prerenderedChildren = [];
+      this.tabs = [];
+      let i = 1;
+      let tabLabel;
+
+      for (const child of this.children) {
+        // Set id and aria-labelledby attributes for each pane for a11y
+        // and remove hidden attribute on first tab.
+        child.id = "w-tab-" + i + "-pane";
+        child.setAttribute("aria-labelledby", "w-tab-" + i);
+        if (i === 1) {
+          child.removeAttribute("hidden");
+        }
+        this.prerenderedChildren.push(child);
+        // Get tab label from pane data-label attribute
+        // and render a tab for each pane.
+        tabLabel = child.getAttribute("data-label");
+        this.tabs.push(this.tabTemplate(i, tabLabel));
+        i++;
+      }
+    }
+
+    return html`
+      <div class="w-tabset" role="tablist" aria-label="${this.label}">
+        ${this.tabs}
+      </div>
+      ${this.prerenderedChildren}
+    `;
+  }
+
+  tabTemplate(i, tabLabel) {
+    let isActive = false;
+    let tabIndex = "-1";
+
+    if (i === 1) {
+      isActive = true;
+      tabIndex = "0";
+    }
+
+    switch (tabLabel) {
+      case "question":
+        tabLabel = "Question " + i;
+        break;
+      case "sample":
+        tabLabel = "Sample " + i;
+        break;
+      case "bare":
+        tabLabel = i;
+        break;
+    }
+
+    return html`
+      <button
+        @click="${this.onClick}"
+        @keydown="${this.onKeydown}"
```

Comment:
Isn't that expected behavior? That is, if you want to move through a large tablist quickly (not that I expect we'll have large tablists in self-assessments), you can just hold down the key--similar to scrolling.

Reply:
yeah I think  what you have is fine actually. I think I have the key repeat speed on my mac turned way up so it goes a bit faster than it would for most other users.